cmd/gardenlet/app/app.go

```diff
@@ -464,15 +477,66 @@ func (g *garden) registerSeed(ctx context.Context, gardenClient client.Client) e
 	})
 }
 
+func (g *garden) migrateAllShootServicesForNetworkPolicies(ctx context.Context, log logr.Logger) error {
+	var (
+		taskFns                  []flow.TaskFn
+		kubeApiServerServiceList = &corev1.ServiceList{}
+	)
+
+	// Kube-Apiserver services
+	if err := g.mgr.GetClient().List(ctx, kubeApiServerServiceList, client.MatchingLabels{v1beta1constants.LabelApp: v1beta1constants.LabelKubernetes, v1beta1constants.LabelRole: v1beta1constants.LabelAPIServer}); err != nil {
+		return err
+	}
+
+	taskFns = append(taskFns, migrationTasksForServices(g.mgr.GetClient(), kubeApiServerServiceList.Items, kubeapiserverconstants.Port)...)
+
+	// VPN-Seed-Server services
+	for _, serviceName := range []string{vpnseedserver.ServiceName, vpnseedserver.ServiceName + "-0", vpnseedserver.ServiceName + "-1"} {
+		serviceList := &corev1.ServiceList{}
+		// Use APIReader here because an index on `metadata.name` is not available in the runtime client.
+		if err := g.mgr.GetAPIReader().List(ctx, serviceList, client.MatchingFieldsSelector{
+			Selector: fields.OneTermEqualSelector(metav1.ObjectNameField, serviceName),
+		}); err != nil {
+			return err
+		}
+
+		taskFns = append(taskFns, migrationTasksForServices(g.mgr.GetClient(), serviceList.Items, vpnseedserver.MetricsPort)...)
+	}
+
+	return flow.Parallel(taskFns...)(ctx)
+}
+
+func migrationTasksForServices(cl client.Client, services []corev1.Service, port int) []flow.TaskFn {
+	var taskFns []flow.TaskFn
+
+	for _, svc := range services {
+		service := svc
+
+		taskFns = append(taskFns, func(ctx context.Context) error {
+			patch := client.MergeFrom(service.DeepCopy())
+
+			metav1.SetMetaDataAnnotation(&service.ObjectMeta, resourcesv1alpha1.NetworkingPodLabelSelectorNamespaceAlias, v1beta1constants.LabelNetworkPolicyShootNamespaceAlias)
+			utilruntime.Must(gardenerutils.InjectNetworkPolicyNamespaceSelectors(&service,
+				metav1.LabelSelector{MatchLabels: map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleIstioIngress}},
+				metav1.LabelSelector{MatchLabels: map[string]string{v1beta1constants.GardenRole: v1beta1constants.GardenRoleExposureClassHandler}}))
```

Comment:
Is it not required to change this as well, now that we use the exposure class name label in other places?

Reply:
Good catch, I changed it as well. Thank you 🙂